### PR TITLE
sci-physics/pythia: add 8.3.12

### PR DESCRIPTION
**Pull request by APN-Pucky**
I enabled python support for this package. The tests numbers have been changed as per https://pythia.org/manuals/pythia8312/SampleMainPrograms.html.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: sci-physics/pythia/pythia-8.3.12.ebuild [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+		https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/${LHA_VER}/MRST2004qed.tar.gz
+	) )
+"
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.com/Pythia8/releases"
+else
+	SRC_URI="https://pythia.org/download/${PN}${MV//./}/${MY_P}.tgz
+	$SRC_URI"
+	KEYWORDS="~amd64"
```
Comment (Nowa-Ammerlaan) [inline sci-physics/pythia/pythia-8.3.12.ebuild:31]: Why are we dropping the `x86` keyword here?

Comment (APN-Pucky) [reply sci-physics/pythia/pythia-8.3.12.ebuild:31]: I was confused first now too, but rivet does not have `~x86`: https://packages.gentoo.org/packages/sci-physics/rivet

Comment (Nowa-Ammerlaan) [reply sci-physics/pythia/pythia-8.3.12.ebuild:31]: A better solution is then to `package.use.mask` the `rivet` USE flag on `x86`. Then we can keep the x86 keyword.

File: sci-physics/pythia/pythia-8.3.12.ebuild [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		https://lhapdfsets.web.cern.ch/lhapdfsets/current/cteq6l1.tar.gz
+		https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/${LHA_VER}/MRST2004qed.tar.gz
+	) )
+"
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.com/Pythia8/releases"
+else
+	SRC_URI="https://pythia.org/download/${PN}${MV//./}/${MY_P}.tgz
+	$SRC_URI"
```
Comment (Nowa-Ammerlaan) [inline sci-physics/pythia/pythia-8.3.12.ebuild:30]:
```suggestion
	${SRC_URI}"
```

**Comment by Nowa-Ammerlaan** [comment]
Thanks, rebased and merged in: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=ed5f074c0ec9d7e30637d47524403f25db80da11

**Pull request closed without merging**